### DISCUSSION TITLE
Overriding strategy loses unmatched keys

### DISCUSSION THREAD

**Discussion by ruimelodev** [toumorokoshi/deepmerge#35] [Q&A] [answered]
I'm trying to setup an overriding strategy, however, when I do so, my base loses all unmatched keys in nxt.

```python
from deepmerge import Merger

my_merger = Merger(
    [
        (list, ["override"]),
        (dict, ["override"]),
        (set, ["override"])
    ],
    ["override"],
    ["override"])

result = my_merger.merge(
    # base
    {"a": [1, 2, 3],
     "b": {"key": "value"},
     "c": True,
     "d": "some text",
     "e": 23.99},

    # nxt
    {"a": [],
     "b": {},
     "d": "some altered text"})
```
Result:
```python
{
    'a': [],
    'b': {},
    'd': 'some altered text'
}
```

Expected result:
```python
{
    'a': [],
    'b': {},
    "c": True,
    'd': 'some altered text',
    "e": 23.99
}
```
If all strategies are set to override, the merge logic is basically returning nxt as is.  
I think that there should be a way to keep base keys even if they don't exist in nxt.

**Comment by toumorokoshi** [comment, marked as answer]
Hi! yeah, I think this is tricky: deepmerge doesn't include the functionality for different behaviors of merging based on nesting.

Since this is just different behavior on the first level, I'd suggest just writing a custom function to run deepmerge on the nested keys. Something like:

```
for key,  value in nxt.items():
    base[key] = my_merge.merge(base.get(key), value)
```
